x-pack/plugins/endpoint/common/generate_data.ts

```diff
@@ -678,6 +678,152 @@ export class EndpointDocGenerator {
     };
   }
 
+  /**
+   * Returns a policy response with all successful configuration and action
+   * statuses.
+   */
+  public generateAllSuccessPolicyResponse(ts = new Date().getTime()): HostPolicyResponse {
```

Comment:
@Paul Tavares not sure if this is the best way to do this (adding another function called generateAllSuccessPolicyResponse in the docGenerator) maybe its better to just add this object directly in the index.test.ts file itself? i just wanted to use all the handy randomizers in the file. 

actually. i might just try to modify the generatePolicyResponse function

Reply:
I think you should only modify the download_model property if it exist and that should prevent the test from being flaky.

Reply:
@parkiino - Suggestion: Add additional (optional) argument to `generatePolicyResponse()` to control the status. Example:

```typescript
public generatePolicyResponse(
  ts = new Date().getTime(),
  actionStatus = HostPolicyResponseActionStatus.success
): HostPolicyResponse {
  //...
}
```

A more complex signature would be to make `actionStatus` argument be a map that instructs the generator as to what the status of specific actions should be - example `actionStatus: { configure_kernel: HostPolicyResponseActionStatus.success }`

The issue is that you are probably trying to get a policy response that drives the many states of the UI, and that' likely going to be difficult without having separate methods.  Perhaps the better way will be to have the test case use the existing `generatePolicyResponse()` and then manipulate the mocked data directly (as the link from @nnamdifrankie shows) to support the different use cases?